### PR TITLE
WT-13548 Relocating log members from WT_CONNECTION_IMPL to log module

### PR DESCRIPTION
**Pull request by bvpvamsikrishna**
Log module data members that are part of the conn struct are relocated to log module.

### CODE REVIEW

**Comment by bvpvamsikrishna** [comment]
Removed the CONN prefix from the values of log_flags.
WT_CONN_LOG_XXX --> WT_LOG_XXX
conn->log_xxx  -----> log_mgr->xxx
Only conn->log_path  ----> log_mgr->log_path (log as prefix is retained)

debug_log_retention_lock, debug_log_cnt are not moved as they are tightly coupled with debug_ckpt related info.

I would like opinions about axing the "log" prefix for the members.
Is the readability is better with this PR, or should I retain "log" as prefix for the members of WT_LOG_MGR;

I will change the name of the structure from __wt_log_mgr --> wt_log_manager.

File: src/log/log.h [hunk 1]
```diff
@@ -125,6 +125,50 @@ struct __wt_txn_printlog_args {
...
+#define WT_LOG_FORCE_DOWNGRADE 0x010u /* Force downgrade */
+#define WT_LOG_INCR_BACKUP 0x020u     /* Incremental backup log required */
+#define WT_LOG_RECOVER_DIRTY 0x040u   /* Recovering unclean */
+#define WT_LOG_RECOVER_DONE 0x080u    /* Recovery completed */
+#define WT_LOG_RECOVER_ERR 0x100u     /* Error if recovery required */
+#define WT_LOG_RECOVER_FAILED 0x200u  /* Recovery failed */
+#define WT_LOG_REMOVE 0x400u          /* Removal is enabled */
+#define WT_LOG_ZERO_FILL 0x800u       /* Manually zero files */
+                                      /* AUTOMATIC FLAG VALUE GENERATION STOP 32 */
+    uint32_t flags;                   /* Global logging configuration */
```
Comment (sueloverso) [inline src/log/log.h:144]: Typically in WT, the flags field is at the end of the structure. I think it helps for readability and consistency with other structures.

File: src/log/log.h [hunk 2]
```diff
@@ -125,6 +125,50 @@ struct __wt_txn_printlog_args {
...
+    bool tid_set;                     /* Log server thread set */
+    WT_CONDVAR *file_cond;            /* Log file thread wait mutex */
+    WT_SESSION_IMPL *file_session;    /* Log file thread session */
+    wt_thread_t file_tid;             /* Log file thread */
+    bool file_tid_set;                /* Log file thread set */
+    WT_CONDVAR *wrlsn_cond;           /* Log write lsn thread wait mutex */
+    WT_SESSION_IMPL *wrlsn_session;   /* Log write lsn thread session */
+    wt_thread_t wrlsn_tid;            /* Log write lsn thread */
+    bool wrlsn_tid_set;               /* Log write lsn thread set */
+    WT_LOG *log;                      /* Logging structure */
```
Comment (sueloverso) [inline src/log/log.h:157]: It feels like the fields from here down should be at the beginning of the structure. The logging structure first, then configuration settings, then the per-thread set of information, then the flags.

**Review by sueloverso — APPROVED** [review]
LGTM. I have a couple small comments about the order of fields in the new struct. It ALL looked like a find-and-replace change. If there were any other changes in there I did not notice them. Please call them out if they exist.

**Comment by sueloverso** [comment]
Regarding your other questions: I prefer what you have done already and removed the `log_` prefix. It is implied in the structure and doesn't add anything helpful to me.

File: src/log/log.h [hunk 3]
```diff
@@ -125,6 +125,50 @@ struct __wt_txn_printlog_args {
...
+    wt_shared uint32_t cursors;       /* Log cursor count */
+    wt_off_t dirty_max;               /* Log dirty system cache max size */
+    wt_off_t file_max;                /* Log file max size */
+    wt_off_t extend_len;              /* file_extend log length */
+    uint32_t force_write_wait;        /* Log force write wait configuration */
+    const char *log_path;             /* Logging path format */
+    uint32_t prealloc;                /* Log file pre-allocation */
+    uint32_t prealloc_init_count;     /* initial number of pre-allocated log files */
+    uint16_t req_max;                 /* Max required log version */
+    uint16_t req_min;                 /* Min required log version */
```
Comment (ajmorton) [inline src/log/log.h:168]: I'm also happy to do this in a separate ticket if it's easier.
We discussed offline about most of these fields being private. Can we either tag them with the `#private` comment or break into private substructs?

**Review by ajmorton — APPROVED** [review]
LGTM. I have a few suggestions but they don't need to be part of this PR. I agree with Sue's points about ordering the fields.

I like the removal of the `log_` prefix from the fields since `log_mgr->field` is easily readable. As discussed offline I'm happy to let developers make that decision when writing the code on a per-field basis

File: src/include/connection.h [hunk 4]
```diff
@@ -761,7 +723,6 @@ struct __wt_connection_impl {
                                       mode before timing out */
 
     wt_off_t data_extend_len; /* file_extend data length */
-    wt_off_t log_extend_len;  /* file_extend log length */
 
 /* AUTOMATIC FLAG VALUE GENERATION START 0 */
```
[the remark this answers is not in the document]
Comment (bvpvamsikrishna) [reply src/include/connection.h:727]: I haven't moved them as mentioned in the initial comment of this PR.
My view is debug_log_retention_lock is the concurrency construct being used for both log and checkpoint debug info. Just moving the debug log fields will make a new cyclic dependency between conn & log due to debug_log_retention_log being part of log module.
I prefer a separate ticket.

File: src/log/log.h (hunk 2 above)
Comment (bvpvamsikrishna) [reply src/log/log.h]: Thank you, I have re organised the members.

File: src/log/log.h (hunk 3 above)
Comment (bvpvamsikrishna) [reply src/log/log.h]: I have added the "private" tag to the members. 
I tried to move the private members to a separate struct, then I realised that it creates a dependency between log.h & log_private.h
As of now log.h doesn't include log_private.h, I prefer to have it this way. 
If we create a dependency of log.h on log_private.h and include it, some other module including "log.h" and accessing a private structure part of "log_private.h" will not be detected by compiler.

Comment (ajmorton) [reply src/log/log.h]: I think we've already lost those compiler protections by #including `log_private.h` from `log_inline.h`, so I don't think it's a big issue.

I'm happy to stick with individual tags, but for future reference we should also be able to create private sub-structs without adding the dependency. 

`s_prototypes` creates the 
```
struct __wt_log_thread;
typedef struct __wt_log_thread WT_LOG_THREAD;
```
forward declarations in `wt_internal.h`. So as long as we declare the sub-structs as pointers
```
struct __wt_log_manager {
    WT_LOG_THREAD *tid
};
```
and only de-reference inside .c files which `#include log_private.h` we'll be ok.

**Pull request merged**